Lesson_13/task1_csv.py

```diff
@@ -0,0 +1,170 @@
+
+    Args:
+        path_file (str): Path to the CSV file.
+        encoding (str, optional): The encoding used to decode the file.
+
+    Returns:
+        list[dict]: List of rows as dictionaries.
+    """
+    try:
+        with open(path_file, mode='r', encoding=encoding) as csv_file:
+            # Read content from the file.
+            content = csv_file.read()
+            # Replace semicolons with comas to delimiters standards.
+            # The rmc.csv file delimiters are displayed as semicolons.
+            content = content.replace(';', ',')
+            # Return content to csv file rows formats
+            csv_file = content.splitlines()
+
+            # Read csv file and returns its data as a list of dictionaries.
+            reader = csv.DictReader(csv_file)
+
+            # Verify file's headers are valid
+            if not reader.fieldnames:
+                raise ValueError(
+                    f'The CSV file "{path_file}" is empty or has no headers.')
+
+            # Check if the file has valid headers
+            data = [{key: value for key, value in row.items() if key.strip()}
+                    for row in reader]
+
+            if not data:
+                raise ValueError(
+                    f'The CSV file {path_file} has no data lines.',
+                )
+
+            return data  # If success.
+
+    except FileNotFoundError:
+        error = f'Error: The specified CSV file "{path_file}" does not exist.'
+        _log.error(error)
+    except ValueError as e:
+        error = f'Data Error: {e}'
+        _log.error(error)
+    except csv.Error as e:
+        error = f'CSV Parsing Error: {e}'
+        _log.error(error)
+    except Exception as e:
+        error = f'An unexpected error occurred: {e}'
+        _log.error(error)
+
+    return error  # If any errors occurred.
+
+
+def write_to_csv(path_file, data, fieldnames, encoding='utf-8'):
+    """Write into CSV file.
+
+    Args:
+        path_file (str): Path to the CSV file.
+        data (list[dict]): List of rows to be written as dictionaries.
+        fieldnames (list[str]): List of column headers for the CSV file.
+        encoding (str, optional): The encoding used to decode the file.
+
+    Returns:
+        str: Path to the CSV file.
+        None: Logs a success message if writing is successful;
+        logs an error otherwise.
+    """
+    try:
+        with open(
+            path_file, mode='w', encoding=encoding, newline='',
+        ) as csv_file:
+            writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+            writer.writeheader()  # Write the header row
+            writer.writerows(data)
+        message = f'Data successfully written to file: "{path_file}".'
+        _log.info(message)
+
+        return path_file  # If success.
+
+    except Exception as e:
+        error = f'An unexpected error occurred: {e}'
+        _log.error(error)
+
+    return error  # If any errors occurred.
```

Comment:
Better to return empty string for error case. Plus this return better to put exactly under exception

Reply:
updated with commit e7ce4b62b49a2daa3850be70c2083336cf6a52d8